src/TypeScript/Rules/RulesRepository.cs

```diff
@@ -72,24 +71,8 @@ public RulesRepository([Import(RuleDefinitionsFilePathContractName)] string type
             .ToList()
             .AsReadOnly();
 
-        private static string GetRuleKey(string eslintRuleKey, IEnumerable<RuleDefinition> rules)
-        {
-            var repoAndKey = rules.FirstOrDefault(x => eslintRuleKey.Equals(x.EslintKey, System.StringComparison.OrdinalIgnoreCase))
+        private static string GetRuleKey(string eslintRuleKey, IEnumerable<RuleDefinition> rules) =>
+            rules.FirstOrDefault(x => eslintRuleKey.Equals(x.EslintKey, System.StringComparison.OrdinalIgnoreCase))
                 ?.RuleKey;
-            return repoAndKey == null ? null : StripRepoPrefix(repoAndKey);
```

Comment:
Out of interest, I started looking at adding `RepoId` as a separate field to `IAnalysisIssue`, expecting it to create a lot of changes.
Unfortunately, it did so I went for the simpler fix of changing the JS rule key to be e.g. `javascript:S123`.

Ticket #2291 ("standardise rule keys") is still open.